api/http_api.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"fmt"
+	"github.com/haveachin/infrared"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+// ListenAndServe StartWebserver Start Webserver if environment variable "api-enable" is set to true
+func ListenAndServe(configPath string, apiBind string) {
+	fmt.Println("Starting WebAPI on " + apiBind)
+	router := chi.NewRouter()
+	router.Use(middleware.Logger)
+
+	router.Post("/proxies", addProxy(configPath))
+	router.Post("/proxies/{fileName}", addProxyWithName(configPath))
+	router.Delete("/proxies/{fileName}", removeProxy(configPath))
+
+	err := http.ListenAndServe(apiBind, router)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+}
+
+func addProxy(configPath string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		rawData, err := ioutil.ReadAll(r.Body)
+		if err != nil || string(rawData) == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		jsonIsValid := checkJSONAndRegister(rawData, "", configPath)
+		if jsonIsValid {
+			w.WriteHeader(http.StatusOK)
+			return
+		} else {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("{'error': 'domainName and proxyTo could not be found'}"))
+			return
+		}
+	}
+}
+
+func addProxyWithName(configPath string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fileName := chi.URLParam(r, "fileName")
+
+		rawData, err := ioutil.ReadAll(r.Body)
+		if err != nil || string(rawData) == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		jsonIsValid := checkJSONAndRegister(rawData, fileName, configPath)
+		if jsonIsValid {
+			w.WriteHeader(http.StatusOK)
+			return
+		} else {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("{'error': 'domainName and proxyTo could not be found'}"))
+			return
+		}
+	}
+}
+
+func removeProxy(configPath string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		file := chi.URLParam(r, "fileName")
+		fmt.Println(file)
+
+		err := os.Remove(configPath + "/" + file)
+		if err != nil {
+			w.WriteHeader(http.StatusNoContent)
+			w.Write([]byte(err.Error()))
+			return
+		}
+	}
+}
+
+// Helper method to check for domainName and proxyTo in a given JSON array
+// If the filename is empty the domain will be used as the filename - files with the same name will be overwritten
+func checkJSONAndRegister(rawData []byte, filename string, configPath string) (successful bool) {
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "infraredTmpConfig_")
+	if err != nil {
+		fmt.Println(err)
+		return false
+	}
+
+	err = os.WriteFile(tmpFile.Name(), rawData, 0644)
+	if err != nil {
+		return false
+	}
+
+	var cfg infrared.ProxyConfig
+	if err := cfg.LoadFromPath(tmpFile.Name()); err != nil {
+		return false
+	}
```

Comment:
This is just unmarshaling the JSON with extra steps. Please just unmarshal ProxyConfig.